src/frontend/src/helpers/websocket_client.ts

```diff
@@ -0,0 +1,469 @@
+import {Coder} from './crypto';
+import * as sjcl from 'sjcl';
+import {History,Settings,Change} from './types';
+
+class Message {
```

Comment:
Easier thing to do here is to use tagged unions -- https://www.typescriptlang.org/docs/handbook/advanced-types.html